src/truncate.jl

```diff
@@ -72,7 +72,7 @@ function truncated(d::UnivariateDistribution, l::T, u::T) where {T <: Real}
     ucdf = exp(logucdf)
 
     # (log)tp = (log) P(l ≤ X ≤ u) where X ∼ d
-    logtp = logsubexp(loglcdf, logucdf)
+    logtp = logdiffcdf(d, u, l)
```

Comment:
This is only correct if the probability mass of `d` at `l` is zero. Hence, while for normal distributions it is correct, for discrete distributions with `pdf(d, l) > 0` it is wrong and should instead be `logaddexp(logdiffcdf(d, u, l), logpdf(d, l))`. Currently, this is handled by excluding these probability masses in `loglcdf` (via `_logcdf_noninclusive`: https://github.com/JuliaStats/Distributions.jl/blob/fd58ce19ba18e15cd0c766893c2928e91ad61ad6/src/censored.jl#L439-L442). A similar fix should be applied here as well:
```suggestion
    logtp = if variate_form(d) === Discrete
        logaddexp(logdiffcdf(d, u, l), logpdf(d, l))
    else
        logdiffcdf(d, u, l)
    end    
```